src/core/qgstiledownloadmanager.cpp

```diff
@@ -325,6 +365,19 @@ void QgsTileDownloadManager::signalQueueModified()
   QMetaObject::invokeMethod( mWorker, &QgsTileDownloadManagerWorker::queueUpdated, Qt::QueuedConnection );
 }
 
+bool QgsTileDownloadManager::isRangeRequest( const QNetworkRequest &request )
+{
+  if ( request.rawHeader( "Range" ).isEmpty() )
+    return false;
+  QRegularExpression regex( "^bytes=\\d+-\\d+$" );
+  QRegularExpressionMatch match = regex.match( QString::fromUtf8( request.rawHeader( "Range" ) ) );
+  return match.hasMatch();
+}
+
+bool QgsTileDownloadManager::isCached( const QNetworkRequest &request )
+{
```

Comment:
We should be also doing caching of range requests if the QNetworkRequest has attributes set to use caching:
- QNetworkRequest::CacheLoadControlAttribute is QNetworkRequest::PreferCache or QNetworkRequest::AlwaysCache (?)
- QNetworkRequest::CacheSaveControlAttribute is true

We should be also setting those attributes appropriately in `QgsCopcPointCloudBlockRequest`:
```cpp
  request.setAttribute( QNetworkRequest::CacheLoadControlAttribute, QNetworkRequest::PreferCache );
  request.setAttribute( QNetworkRequest::CacheSaveControlAttribute, true );
```